code/script.js

```diff
@@ -269,57 +297,117 @@ const filterCharacters = (keep) => {
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `Oh nooooo, the person doesn't wear ${value}! Remove all people that wears ${value}`
       )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+
     }
   } else if (category === 'other') {
-    // Similar to the one above
+    if (keep) {
+      alert(
+        `Siiii eso es!!, the person wears ${value}! Keep all people that wears ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+  
+    
+    } else {
+      alert(
+        `Ay no!, the person doesn't wear ${value}! Remove all people that wears ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      
+    }  
+
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value} ${category} Keep all people with ${value} ${category}.`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)    
     }
   }
+ 
+  generateBoard();
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
+// store the interaction from the player in a variable.
   // If the player wants to guess, invoke the checkMyGuess function.
+const guess = (personToConfirm) => {
+  let guessed = confirm (`hmmmm are you sure?, really really sure that ${personToConfirm} is your guess?`)
+  if(guessed){
+    let personToCheck = personToConfirm; //variable that saves the character the player has shosen
+checkMyGuess(personToCheck)
```

Comment:
Think it should say in "checkMyGuess(personToCheck)", instead write "checkMyGuess(personToConfirm)".